src/manifolds/HyperbolicHyperboloid.jl

```diff
@@ -226,7 +226,11 @@ d_{\mathcal H^n}(p,q) = \operatorname{acosh}( - ⟨p, q⟩_{\mathrm{M}}),
 where $⟨\cdot,\cdot⟩_{\mathrm{M}}$ denotes the [`MinkowskiMetric`](@ref) on the embedding,
 the [`Lorentz`](@ref)ian manifold.
 """
-distance(::Hyperbolic, p, q) = acosh(max(-minkowski_metric(p, q), 1.0))
+function distance(::Hyperbolic, p, q)
+    w = q - p
+    m = sqrt(max(0.0, minkowski_metric(w, w)))
+    return 2 * asinh(0.5 * m)
```

Comment:
```suggestion
    return 2 * asinh(m / 2)
```
Minor point: `m / 2` doesn't force promotion to `Float64`